### PR TITLE
Audio15

### PR DESCRIPTION
**Pull request by ShFil119**
Implements 5 more methods plus makes some cleanup.

### CODE REVIEW

File: src/core/Fire.h [hunk 1]
```diff
@@ -28,8 +28,8 @@ class CFireManager
 		MAX_FIREMEN_ATTENDING = 2,
 	};
 	uint32 m_nTotalFires;
-	CFire m_aFires[NUM_FIRES];
 public:
+	CFire m_aFires[NUM_FIRES];
```
Comment (ShFil119) [inline src/core/Fire.h]: or better to add getter?

File: src/audio/AudioManager.cpp [hunk 2]
```diff
@@ -3172,11 +3173,74 @@ cAudioManager::ProcessCarBombTick(cVehicleParams *params)
...
+	} else if(params->m_nIndex == DODO) {
+		AddPlayerCarSample(105, 17000, SFX_CESNA_IDLE, 0, 52, 1);
+	} else if(params->m_fDistance < 40000.f) {
+		CalculateDistance(params->m_bDistanceCalculated, params->m_fDistance);
+		m_sQueueSample.m_bVolume = ComputeVolume(80, 200.f, m_sQueueSample.m_fDistance);
+		if(m_sQueueSample.m_bVolume) {
+			m_sQueueSample.m_counter = 52;
+			m_sQueueSample.m_nSampleIndex = SFX_CESNA_IDLE;
+			m_sQueueSample.m_bBankIndex = 0;
+			m_sQueueSample.m_bIsDistant = 0;
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: = false

File: src/audio/AudioManager.cpp [hunk 3]
```diff
@@ -3172,11 +3173,74 @@ cAudioManager::ProcessCarBombTick(cVehicleParams *params)
...
+		}
+	} else if(params->m_nIndex == DODO) {
+		AddPlayerCarSample(105, 17000, SFX_CESNA_IDLE, 0, 52, 1);
+	} else if(params->m_fDistance < 40000.f) {
+		CalculateDistance(params->m_bDistanceCalculated, params->m_fDistance);
+		m_sQueueSample.m_bVolume = ComputeVolume(80, 200.f, m_sQueueSample.m_fDistance);
+		if(m_sQueueSample.m_bVolume) {
+			m_sQueueSample.m_counter = 52;
+			m_sQueueSample.m_nSampleIndex = SFX_CESNA_IDLE;
+			m_sQueueSample.m_bBankIndex = 0;
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: = SAMPLEBANK_MAIN

File: src/audio/AudioManager.cpp [hunk 4]
```diff
@@ -3691,11 +3874,311 @@ cAudioManager::ProcessFrontEnd()
...
+											}
+										}
+										break;
+									default: continue;
+									}
+								}
+								continue;
+							}
+							if(m_anRandomTable[1] & 1) {
+								sampleIndex = m_anRandomTable[2] % 5u + SFX_COL_CAR_1;
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: 5

File: src/audio/AudioManager.cpp [hunk 5]
```diff
@@ -3691,11 +3874,311 @@ cAudioManager::ProcessFrontEnd()
...
+							} else {
+								sampleIndex =
+								    m_anRandomTable[2] % 6u + SFX_COL_CAR_PANEL_1;
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: 6

File: src/audio/AudioManager.cpp [hunk 6]
```diff
@@ -3691,11 +3874,311 @@ cAudioManager::ProcessFrontEnd()
...
+													m_sQueueSample
+													    .m_nFrequency =
+													    6735;
+												} else if(
+												    m_asAudioEntities
+												        [m_sQueueSample
+												             .m_nEntityIndex]
+												            .m_awAudioEvent
+												                [j] ==
+												    69) {
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: SOUND_GARAGE_DOOR_OPENED

File: src/audio/AudioManager.cpp [hunk 7]
```diff
@@ -3691,11 +3874,311 @@ cAudioManager::ProcessFrontEnd()
...
+											if(iSound < 32) iSound = 32;
+											m_sQueueSample
+											    .m_bRequireReflection = 1;
+											AddSampleToRequestedQueue();
+										}
+									}
+									break;
+								default: continue;
+								}
+							}
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: redundant loop

File: src/audio/AudioManager.cpp [hunk 8]
```diff
@@ -3691,11 +3874,311 @@ cAudioManager::ProcessFrontEnd()
...
+									m_sQueueSample.m_nLoopEnd = -1;
+									m_sQueueSample.m_counter = iSound++;
+									if(iSound < 32) iSound = 32;
+									m_sQueueSample.m_bRequireReflection = 1;
+									AddSampleToRequestedQueue();
+								}
+							}
+							break;
+						default: continue;
+						}
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: redundant loop

File: src/audio/AudioManager.cpp [hunk 9]
```diff
@@ -7064,7 +7547,7 @@ cAudioManager::ProcessVehicleHorn(cVehicleParams *params)
 				if(automobile->m_nCarHornTimer > 44) automobile->m_nCarHornTimer = 44;
 				if(automobile->m_nCarHornTimer == 44)
 					automobile->field_22D =
-					    (LOBYTE(m_nTimeOfRecentCrime) + LOBYTE(m_sQueueSample.m_nEntityIndex)) & 7;
+					    (LOBYTE(m_FrameCounter) + LOBYTE(m_sQueueSample.m_nEntityIndex)) & 7;
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: LOBYTE -> (uint8)

File: src/audio/AudioManager.cpp [hunk 10]
```diff
@@ -7739,8 +8222,34 @@ cAudioManager::SetSpeakerConfig(int32 conf) const
...
+cAudioManager::SetupJumboEngineSound(uint8 vol, int32 freq)
+{
+	if(m_sQueueSample.m_fDistance >= 180.f) return 0;
+
+	uint8 emittingVol = vol - gJumboVolOffsetPercentage / 100;
+	m_sQueueSample.m_bVolume = ComputeVolume(emittingVol, 180.f, m_sQueueSample.m_fDistance);
+	if(m_sQueueSample.m_bVolume) {
+		m_sQueueSample.m_counter = 3;
+		m_sQueueSample.m_nSampleIndex = SFX_JUMBO_ENGINE;
+		m_sQueueSample.m_bBankIndex = 0;
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: = SAMPLEBANK_MAIN

File: src/audio/AudioManager.cpp [hunk 11]
```diff
@@ -7739,8 +8222,34 @@ cAudioManager::SetSpeakerConfig(int32 conf) const
 	SampleManager.SetSpeakerConfig(conf);
 }
 
-WRAPPER
-bool cAudioManager::SetupJumboEngineSound(uint8, int32) { EAXJMP(0x56F140); }
+bool
+cAudioManager::SetupJumboEngineSound(uint8 vol, int32 freq)
+{
+	if(m_sQueueSample.m_fDistance >= 180.f) return 0;
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: return false

File: src/audio/AudioManager.cpp [hunk 12]
```diff
@@ -7739,8 +8222,34 @@ cAudioManager::SetSpeakerConfig(int32 conf) const
...
+
+	uint8 emittingVol = vol - gJumboVolOffsetPercentage / 100;
+	m_sQueueSample.m_bVolume = ComputeVolume(emittingVol, 180.f, m_sQueueSample.m_fDistance);
+	if(m_sQueueSample.m_bVolume) {
+		m_sQueueSample.m_counter = 3;
+		m_sQueueSample.m_nSampleIndex = SFX_JUMBO_ENGINE;
+		m_sQueueSample.m_bBankIndex = 0;
+		m_sQueueSample.m_bIsDistant = 0;
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: = false

File: src/audio/AudioManager.cpp [hunk 13]
```diff
@@ -7739,8 +8222,34 @@ cAudioManager::SetSpeakerConfig(int32 conf) const
...
+		m_sQueueSample.m_nFrequency = freq;
+		m_sQueueSample.m_nLoopCount = 0;
+		m_sQueueSample.m_bEmittingVolume = emittingVol;
+		m_sQueueSample.m_nLoopStart = SampleManager.GetSampleLoopStartOffset(m_sQueueSample.m_nSampleIndex);
+		m_sQueueSample.m_nLoopEnd = SampleManager.GetSampleLoopEndOffset(m_sQueueSample.m_nSampleIndex);
+		m_sQueueSample.field_48 = 4.0;
+		m_sQueueSample.m_fSoundIntensity = 180.0f;
+		m_sQueueSample.field_56 = 0;
+		m_sQueueSample.field_76 = 4;
+		m_sQueueSample.m_bReverbFlag = 1;
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: = true

File: src/audio/AudioManager.cpp [hunk 14]
```diff
@@ -7739,8 +8222,34 @@ cAudioManager::SetSpeakerConfig(int32 conf) const
...
+		m_sQueueSample.m_nSampleIndex = SFX_JUMBO_ENGINE;
+		m_sQueueSample.m_bBankIndex = 0;
+		m_sQueueSample.m_bIsDistant = 0;
+		m_sQueueSample.field_16 = 1;
+		m_sQueueSample.m_nFrequency = freq;
+		m_sQueueSample.m_nLoopCount = 0;
+		m_sQueueSample.m_bEmittingVolume = emittingVol;
+		m_sQueueSample.m_nLoopStart = SampleManager.GetSampleLoopStartOffset(m_sQueueSample.m_nSampleIndex);
+		m_sQueueSample.m_nLoopEnd = SampleManager.GetSampleLoopEndOffset(m_sQueueSample.m_nSampleIndex);
+		m_sQueueSample.field_48 = 4.0;
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: 4.0f

File: src/audio/AudioManager.cpp [hunk 15]
```diff
@@ -7739,8 +8222,34 @@ cAudioManager::SetSpeakerConfig(int32 conf) const
...
+		m_sQueueSample.m_fSoundIntensity = 180.0f;
+		m_sQueueSample.field_56 = 0;
+		m_sQueueSample.field_76 = 4;
+		m_sQueueSample.m_bReverbFlag = 1;
+		m_sQueueSample.m_bRequireReflection = 0;
```
Comment (Sergeanur) [inline src/audio/AudioManager.cpp]: = false

File: src/control/Pickups.cpp [hunk 16]
```diff
@@ -704,7 +704,7 @@ CPickups::DoPickUpEffects(CEntity *entity)
 			aWeaponReds[colorId] * colorModifier, aWeaponGreens[colorId] * colorModifier, aWeaponBlues[colorId] * colorModifier,
 			4.0f, 1.0f, 40.0f, false, 0.0f);
 
-		float radius = (double)(rand() & 0xF) * 0.1 + 3.0;
+		float radius = (double)(rand() & 0xF) * 0.1 + 3.0f;
```
Comment (Sergeanur) [inline src/control/Pickups.cpp]: 0.1f

File: src/control/Pickups.cpp [hunk 17]
```diff
@@ -704,7 +704,7 @@ CPickups::DoPickUpEffects(CEntity *entity)
 			aWeaponReds[colorId] * colorModifier, aWeaponGreens[colorId] * colorModifier, aWeaponBlues[colorId] * colorModifier,
 			4.0f, 1.0f, 40.0f, false, 0.0f);
 
-		float radius = (double)(rand() & 0xF) * 0.1 + 3.0;
+		float radius = (double)(rand() & 0xF) * 0.1 + 3.0f;
 		CPointLights::AddLight(CPointLights::LIGHT_POINT, pos, CVector(0.0f, 0.0f, 0.0f), radius, aWeaponReds[colorId] * modifiedSin / 256.0f, aWeaponGreens[colorId] * modifiedSin / 256.0f, aWeaponBlues[colorId] * modifiedSin / 256.0f, CPointLights::FOG_NONE, true);
 		float size = (double)(rand() & 0xF) * 0.0005 + 0.6;
```
Comment (Sergeanur) [inline src/control/Pickups.cpp]: 0.0005f + 0.6f

File: src/control/Bridge.cpp [hunk 18]
```diff
@@ -60,32 +60,32 @@ void CBridge::Update()
 		if (timeElapsed < 10000)
 		{
 			State = STATE_LIFT_PART_MOVING_DOWN;
-			liftHeight = 25.0 - timeElapsed / 10000.0 * 25.0;
+			liftHeight = 25.0 - timeElapsed / 10000.0 * 25.0f;
```
Comment (Sergeanur) [inline src/control/Bridge.cpp]: first 25.0f

File: src/control/Pickups.cpp [hunk 19]
```diff
@@ -704,9 +704,9 @@ CPickups::DoPickUpEffects(CEntity *entity)
 			aWeaponReds[colorId] * colorModifier, aWeaponGreens[colorId] * colorModifier, aWeaponBlues[colorId] * colorModifier,
 			4.0f, 1.0f, 40.0f, false, 0.0f);
 
-		float radius = (double)(rand() & 0xF) * 0.1 + 3.0;
+		float radius = (rand() & 0xF) * 0.1f + 3.0f;
 		CPointLights::AddLight(CPointLights::LIGHT_POINT, pos, CVector(0.0f, 0.0f, 0.0f), radius, aWeaponReds[colorId] * modifiedSin / 256.0f, aWeaponGreens[colorId] * modifiedSin / 256.0f, aWeaponBlues[colorId] * modifiedSin / 256.0f, CPointLights::FOG_NONE, true);
-		float size = (double)(rand() & 0xF) * 0.0005 + 0.6;
+		float size = (rand() & 0xF) * 0.0005f + 0.6f;
 		CCoronas::RegisterCorona( (uintptr)entity,
```
Comment (erorcun) [inline src/control/Pickups.cpp]: All rand()'s are inlined CGeneral::GetRandomNumber according to aap, you should use that

File: src/core/common.h [hunk 20]
```diff
@@ -4,7 +4,6 @@
 #define _USE_MATH_DEFINES
 #pragma warning(disable: 4244)	// int to float
 #pragma warning(disable: 4800)	// int to bool
-#pragma warning(disable: 4305)	// double to float
```
Comment (erorcun) [inline src/core/common.h]: why?

Comment (ShFil119) [reply src/core/common.h]: now it'll remind if we don't use our math wraps or double number.

Comment (erorcun) [reply src/core/common.h]: ok

File: src/audio/AudioManager.h [hunk 21]
```diff
@@ -2,6 +2,7 @@
 
 #include "DMAudio.h"
 #include "common.h"
+#include "core/config.h"
```
Comment (erorcun) [inline src/audio/AudioManager.h]: only "config.h" pls

Comment (ShFil119) [reply src/audio/AudioManager.h]: done

**Pull request merged**